### PR TITLE
[REG2.068.1] Issue 15030 - ICE with recursive delegate, -unittest, and std.range

### PR DESCRIPTION
**Pull request by 9rnsr**
https://issues.dlang.org/show_bug.cgi?id=15030

Refer the detailed explanation in `test/runnable/ice15030.d`.

### CODE REVIEW

File: src/glue.c [hunk 1]
```diff
@@ -804,23 +821,15 @@ void FuncDeclaration_toObjFile(FuncDeclaration *fd, bool multiobj)
...
-            if (UnitTestDeclaration *udp = fdp->isUnitTestDeclaration())
-            {
-                udp->deferredNested.push(fd);
-                return;
-            }
-        }
+        /* Can't do unittest's out of order, they are order dependent in that their
+         * execution is done in lexical order.
+         */
+        udp->deferredNested.push(fd);
```
Comment (MartinNowak) [inline src/glue.c]: How comes that needed functions in unittests go to the object file earlier? Is this b/c of the appendToModuleMember or so?

Comment (9rnsr) [inline src/glue.c]: The direct cause is the operations in `appendToModuleMember`. but there's more big underlying issue in current dmd internals for nested template instances.

Comment (9rnsr) [inline src/glue.c]: The second commit is an (incomplete) fix for the "big issue", but it doesn't work.
So, in the third commit, I adjusted `appendToModuleMember` in order to fix the order of instantiated functions generation instead.

**Comment by MartinNowak** [comment]
This fixed another order related bug, [Issue 15032 – [REG2.068.1] coverage output is discreted around the calls to map(), canFind(), filter()](https://issues.dlang.org/show_bug.cgi?id=15032).
It's pretty obvious that we need a proper dependency pipeline rather than trying to explicitly track `codegen find -> codegen main -> coverage symbol mod_b`.

**Pull request merged**